### ISSUE TITLE
CaptureRequest not handled in case that user captures before the request flow starts collection 

### ISSUE THREAD

**Issue by ssttkkl** [PatilShreyas/Capturable#202]
A case of that is to capture immediately after the page is shown:

```kotlin
        // The content to be captured ⬇️
        Ticket(modifier = Modifier.capturable(captureController))

        Spacer(modifier = Modifier.size(32.dp))

        // Captures ticket bitmap on click
        Button(
            onClick = {
                uiScope.launch {
                    ticketBitmap = captureController.captureAsync().await()
                }
            }
        ) {
            Text("Preview Ticket Image")
        }

        LaunchedEffect(Unit) {
            ticketBitmap = captureController.captureAsync().await()
        }
```

In such case, captureRequest will be emitted (in CaptureController#captureAsync) before CapturableModifierNode starts to collect the captureRequests flow (in CapturableModifierNode#observeCaptureRequestsAndServe), so this request will never be handled.

Reproducer: https://github.com/ssttkkl/Capturable/commit/57f4d9c9e4de70b16740c24e46041056856c596d

**Comment by ssttkkl** [comment]
A simple solution is make the observing job launch in dispatcher except the Main dispatcher. I tried to fix it and maybe I can make a PR to solve this issue. https://github.com/ssttkkl/Capturable/commit/7bad2cfaf56eb59cd5c95834f489e69fb0a03346

**Comment by ssttkkl** [comment]
A simple solution is to set `extraBufferCapacity = 0` and use suspend `emit` method, so that the emission will suspend until the flow collection starts. 

I tried to fix it: https://github.com/ssttkkl/Capturable/commit/7bad2cfaf56eb59cd5c95834f489e69fb0a03346 An extra coroutineScope was introduced to run the emission and wait for result async.

**Comment by PatilShreyas** [comment]
Thanks for reporting the issue. Will look into this, I would not want to introduce another CoroutineScope here as it'll serve concurrent requests as well which we don't want.

**Comment by PatilShreyas** [comment]
Fixed in v3.0.0
